### PR TITLE
refactor: factorize some updates in ITaskTable

### PR DESCRIPTION
**Pull request by aneojgurhem**

### CODE REVIEW

File: Common/src/Storage/TaskTableExtensions.cs [hunk 1]
```diff
@@ -18,23 +18,27 @@
 using System;
 using System.Collections.Generic;
 using System.Linq.Expressions;
+using System.Runtime.Intrinsics.Arm;
```
Comment (lemaitre-aneo) [inline Common/src/Storage/TaskTableExtensions.cs:21]: Why do you have this?

File: Common/src/Storage/TaskTableExtensions.cs [hunk 2]
```diff
@@ -179,4 +182,100 @@ public static class TaskTableExtensions
...
+  /// <returns>
+  ///   The number of updated tasks
+  /// </returns>
+  public static async Task<long> UpdateAllTaskStatusAsync(this ITaskTable   taskTable,
+                                                          TaskFilter        filter,
+                                                          TaskStatus        status,
+                                                          CancellationToken cancellationToken = default)
+  {
+    if (filter.Included != null && (filter.Included.Statuses.Contains(TaskStatus.Completed) || filter.Included.Statuses.Contains(TaskStatus.Cancelled) ||
+                                    filter.Included.Statuses.Contains(TaskStatus.Error)))
```
Comment (lemaitre-aneo) [inline Common/src/Storage/TaskTableExtensions.cs:202]:
```suggestion
    if (filter.Included != null && (filter.Included.Statuses.Contains(TaskStatus.Completed) || filter.Included.Statuses.Contains(TaskStatus.Cancelled) ||
                                    filter.Included.Statuses.Contains(TaskStatus.Error) || filter.Included.Statuses.Contains(TaskStatus.Retried)))
```

File: Common/src/Storage/TaskTableExtensions.cs [hunk 3]
```diff
@@ -179,4 +182,100 @@ public static class TaskTableExtensions
...
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  ///   The number of task matched
+  /// </returns>
+  public static async Task<long> CancelTaskAsync(this ITaskTable     taskTable,
+                                                 ICollection<string> taskIds,
+                                                 CancellationToken   cancellationToken = default)
+    => await taskTable.UpdateManyTasks(data => taskIds.Contains(data.TaskId) &&
+                                               !(data.Status == TaskStatus.Cancelled || data.Status == TaskStatus.Cancelling || data.Status == TaskStatus.Error ||
+                                                 data.Status == TaskStatus.Completed),
```
Comment (lemaitre-aneo) [inline Common/src/Storage/TaskTableExtensions.cs:252]:
```suggestion
                                               !(data.Status == TaskStatus.Cancelled || data.Status == TaskStatus.Cancelling || data.Status == TaskStatus.Error ||
                                                 data.Status == TaskStatus.Completed || data.Status == TaskStatus.Retried),
```

File: Common/tests/TestBase/TaskTableTestBase.cs [hunk 4]
```diff
@@ -554,20 +554,6 @@ public async Task CancelSessionAsyncShouldSucceed()
...
-    if (RunTests)
-    {
-      Assert.ThrowsAsync<SessionNotFoundException>(async () =>
-                                                   {
-                                                     await TaskTable!.CancelSessionAsync("NonExistingSessionId",
-                                                                                         CancellationToken.None)
-                                                                     .ConfigureAwait(false);
-                                                   });
-    }
-  }
```
Comment (lemaitre-aneo) [inline Common/tests/TestBase/TaskTableTestBase.cs:569]: Why remove this test?

Comment (aneojgurhem) [reply Common/tests/TestBase/TaskTableTestBase.cs:569]: This method does not throw `SessionNotFoundException` anymore.

File: Common/src/Storage/TaskTableExtensions.cs [hunk 5]
```diff
@@ -22,19 +22,22 @@
...
-    => taskTable.UpdateAllTaskStatusAsync(filter,
-                                          TaskStatus.Cancelling,
-                                          cancellationToken);
+  public static async Task<int> CancelTasks(this ITaskTable   taskTable,
+                                            TaskFilter        filter,
+                                            CancellationToken cancellationToken = default)
+    => (int)await taskTable.UpdateAllTaskStatusAsync(filter,
+                                                     TaskStatus.Cancelling,
+                                                     cancellationToken)
+                           .ConfigureAwait(false);
```
Comment (lemaitre-aneo) [inline Common/src/Storage/TaskTableExtensions.cs:40]: This looks like a duplicate of CancelTasksAsync

Comment (aneojgurhem) [reply Common/src/Storage/TaskTableExtensions.cs:40]: It is not the same filtering in input. TaskFilter is a gRPC object we convert to expression

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**